mne/epochs.py

```diff
@@ -2219,46 +2298,67 @@ def _compare_epochs_infos(info1, info2, ind):
         raise ValueError('SSP projectors in epochs files must be the same')
 
 
-def concatenate_epochs(epochs_list):
-    """Concatenate a list of epochs into one epochs object
-
-    Parameters
-    ----------
-    epochs_list : list
-        list of Epochs instances to concatenate (in order).
-
-    Returns
-    -------
-    epochs : instance of Epochs
-        The result of the concatenation (first Epochs instance passed in).
-
-    Notes
-    -----
-    .. versionadded:: 0.9.0
-    """
+def _concatenate_epochs(epochs_list, read_file=False):
+    """Auxiliary function for concatenating epochs."""
     out = epochs_list[0]
     data = [out.get_data()]
     events = [out.events]
     drop_log = cp.deepcopy(out.drop_log)
     event_id = cp.deepcopy(out.event_id)
+    selection = out.selection
     for ii, epochs in enumerate(epochs_list[1:]):
         _compare_epochs_infos(epochs.info, epochs_list[0].info, ii)
         if not np.array_equal(epochs.times, epochs_list[0].times):
             raise ValueError('Epochs must have same times')
 
+        if epochs.baseline != epochs_list[0].baseline:
+            raise ValueError('Baseline must be same for all epochs')
+
         data.append(epochs.get_data())
         events.append(epochs.events)
-        drop_log.extend(epochs.drop_log)
+        selection = np.concatenate((selection, epochs.selection))
+        if read_file:
+            for k, (a, b) in enumerate(zip(drop_log, epochs.drop_log)):
+                if a == ['IGNORED'] and b != ['IGNORED']:
+                    drop_log[k] = b
+        else:
+            drop_log.extend(epochs.drop_log)
```

Comment:
yeah, I don't see a use case when you want to merge epochs which don't come from the same source ...